src/mmw/js/src/modeling/models.js

```diff
@@ -502,6 +502,16 @@ var ProjectModel = Backbone.Model.extend({
         return url;
     },
 
+    getSnapshot: function() {
+        // Stringify and parse the project to convert all Backbone models and
+        // collections to JSON objects and arrays.
+        var snapshot = JSON.parse(JSON.stringify(this));
+
+        snapshot.snapshot_at = (new Date()).toISOString();
```

Comment:
How much of a problem is it that the snapshot date is coming from the client? We can't 100% trust the user's clock.

Reply:
Hmm, good point. I'll look into doing this on the server side.